packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -235,6 +213,8 @@ class InnerBlocks extends Component {
 const ComposedInnerBlocks = compose( [
 	withViewportMatch( { isSmallScreen: '< medium' } ),
 	withBlockEditContext( ( context ) => pick( context, [ 'clientId' ] ) ),
+	withRegistry,
+	withBlockEntitySync, // Required to be after withRegistry
```

Comment:
There's a good  opportunity to refactor InnerBlocks as a functional component after this PR to avoid  this HoC. I find components with a lot of HoCs a  bit hard to follow.

Reply:
I agree. I can definitely do that ;)

Reply:
done ;)